PowerPointLabs/PowerPointLabs/ELearningLab/Service/FunctionService/CalloutService.cs

```diff
@@ -52,9 +52,18 @@ private static Shape InsertCalloutShapeToSlide(PowerPointSlide slide, string cal
                 templatedShape = shape;
                 return shape;
             }
-            if (templatedShape != null && slide.ContainShape(templatedShape))
+            if (templatedShape != null)
             {
-                return ShapeUtility.InsertTemplatedShapeToSlide(slide, templatedShape, shapeName, calloutText);
+                Shape copiedShape;
```

Comment:
Is it better if the `if` branches are inverted? Using templatedShape == null as a guard clause with a 1 line return, then have the try catch block afterwards?

Reply:
You're right! That's much better. I checked for `templatedShape == null` in `ShapeUtility.InsertTemplatedShapeToSlide` method instead, and added try-catch block for that method when it was called.